supernode/daemon/mgr/scheduler/manager.go

```diff
@@ -228,6 +231,26 @@ func (sm *Manager) tryGetPID(ctx context.Context, taskID string, pieceNum int, s
 			continue
 		}
 
+		// if the v is in the blackList, try the next one.
+		blackInfo, err := sm.progressMgr.GetBlackInfoByPeerID(ctx, srcPID)
+		if err != nil && !errortypes.IsDataNotFound(err) {
+			logrus.Warnf("scheduler: failed to get blackInfo for peerID %s: %v", peerIDs[i], err)
+			continue
+		}
+		if isExistInMap(blackInfo, peerIDs[i]) {
+			// if the peer is in blank list and dfgetTask finishes(this peer may be off without report server down successfully), server down time by supernode.
+			cid, err := sm.dfgetTask.GetCIDByPeerIDAndTaskID(ctx, peerIDs[i], taskID)
+			if err != nil {
+				logrus.Warnf("failed to get cID by peerID %s and TaskID %s, err: %v", peerIDs[i], taskID, err)
+				continue
+			}
+			dfgetTask, err := sm.dfgetTask.Get(ctx, cid, taskID)
+			if err == nil && dfgetTask.Status == types.DfGetTaskStatusSUCCESS {
+				sm.progressMgr.UpdatePeerServiceDown(ctx, peerIDs[i])
```

Comment:
Firstly, we should understand the relationship between peer and dfgetTask. In fact, one peer  corresponds to multi dfgetTasks. So  we should also confirm if any other dfgetTask is running before we  judge a peer might go offline.

Reply:
oh,I ignore that,I will do with it

Reply:
if a peer is off when dowenload,the dfgetTask will be unfinished,how we judge it?

Reply:
I mean a peer which crashes  will have some dead dfgetTask,it hard to judge it ,can you have some good idea.

Reply:
If a dfgetTask has not been updated for a long time with running state, and then we shall think this task has failed. WDYT? @yunfeiyanggzq @lowzj

Reply:
have we recorded the  latest dfget access time in Dragonfly?if not ,do you think we need to add the time

Reply:
In fact, we have not recorded the latest dfget access time. And that's exactly what  we should do.